pkg/component/gardener/discoveryserver/discoveryserver.go

```diff
@@ -0,0 +1,190 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package discoveryserver
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Masterminds/semver/v3"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	operatorv1alpha1 "github.com/gardener/gardener/pkg/apis/operator/v1alpha1"
+	"github.com/gardener/gardener/pkg/component"
+	operatorclient "github.com/gardener/gardener/pkg/operator/client"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+	secretsutils "github.com/gardener/gardener/pkg/utils/secrets"
+	secretsmanager "github.com/gardener/gardener/pkg/utils/secrets/manager"
+)
+
+const (
+	// ManagedResourceNameRuntime is the name of the ManagedResource for the runtime resources.
+	ManagedResourceNameRuntime = "gardener-discovery-server-runtime"
+	// ManagedResourceNameVirtual is the name of the ManagedResource for the virtual resources.
+	ManagedResourceNameVirtual = "gardener-discovery-server-virtual"
```

Comment:
I think you can unexport them?

Reply:
Yes, but this way it is consistent with the other components. See example for the dashboard https://github.com/gardener/gardener/blob/6eb518fb0e68f93e32cd9615a9cb399242aefca5/pkg/component/gardener/dashboard/dashboard.go#L28-L31

It would probably make sense to keep it consistent across implementations.

Reply:
Yes, that's a left-over from previous times where the `garden-care` controller needed to know the exact MR names. Meanwhile, if they are not used outside, we can unexport them everywhere. Feel free to harmonize this if you want :)